### PR TITLE
[device/dell] Add z9100 pfc T0 support

### PR DESCRIPTION
**Pull request by vharish02**
This commit adds new code to support z9100 PFC T0 support with
broadcom recommended MMU settings for PFC feature.

Unit tested the setting by loading sonic-broadcom.bin and checking
the hardware for the values from the JSON file. The T0 configs supports
fan-out of 100G ports on Z9100. Added new config.bcm for fanout of 100G
ports and tested the fanout by sending traffic using bcmcmd, new file
will be copied to /etc/bcm/th-z9100-8x100-48x50G.config.bcm. This file
has to be packaged with libsaibcm library. Will send the file to Ying.

Signed-off-by: Harish Venkatraman <Harish_Venkatraman@dell.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vharish02** [comment]
@yxieca Can you please review

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Ethernet6       55,56           FiftyGigE1/2/2         2      50000
+Ethernet8       57,58           FiftyGigE1/3/1         3      50000
+Ethernet10      59,60           FiftyGigE1/3/2         3      50000
+Ethernet12      61,62           FiftyGigE1/4/1         4      50000
+Ethernet14      63,64           FiftyGigE1/4/2         4      50000
+Ethernet16      65,66           FiftyGigE1/5/1         5      50000
+Ethernet18      67,68           FiftyGigE1/5/2         5      50000
+Ethernet20      69,70           FiftyGigE1/6/1         6      50000
+Ethernet22      71,72           FiftyGigE1/6/2         6      50000
+Ethernet24      73,74,75,76     hundredGigE1/7         7      100000
```
Comment (prsunny) [inline device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini]: Just a suggestion to use one convention, either all alias name starts with lowercase or all with uppercase. Also can you check if for FiftyGig, it is `1/1/2` or `1/1/3`

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/sai.profile [hunk 2]
```diff
@@ -0,0 +1 @@
+SAI_INIT_CONFIG_FILE=/etc/bcm/th-z9100-8x100-48x50G.config.bcm
```
Comment (prsunny) [inline device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/sai.profile]: Just an observation, rest of the config files specify `G` as in `8x100G`

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini (hunk 1 above)
Comment (vharish02) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini]: Sure will change

Comment (vharish02) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini]: I followed the convention like what was done in FTOS where it is fiftyGiGE1/1/1 and 1/1/2.

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/sai.profile (hunk 2 above)
Comment (vharish02) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/sai.profile]: ok will change accordingly

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini (hunk 1 above)
Comment (yxieca) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini]: The point is at the first letter, please use lower case as you suggested.

File: device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Ethernet96      117,118         FiftyGigE1/25/1        25     50000
+Ethernet98      119,120         FiftyGigE1/25/2        25     50000
+Ethernet100     113,114         FiftyGigE1/26/1        26     50000
+Ethernet102     115,116         FiftyGigE1/26/2        26     50000
+Ethernet104     125,126,127,128 hundredGigE1/27        27     100000
+Ethernet108     121,122,123,124 hundredGigE1/28        28     100000
+Ethernet112     5,6,7,8         hundredGigE1/29        29     100000
+Ethernet116     1,2,3,4         hundredGigE1/30        30     100000
+Ethernet120     13,14           FiftyGigE1/31/1        31     50000
+Ethernet122     15,16           FiftyGigE1/31/1        31     50000
```
[the remark this answers is not in the document]
Comment (yxieca) [reply device/dell/x86_64-dell_z9100_c2538-r0/Force10-Z9100-T0/port_config.ini]: Have you followed the porting guide, particularly testing on your target device and making sure that all ports come up online.

**Pull request merged**